markdown/guide/addons.md

```diff
@@ -45,6 +49,16 @@ module.exports = {
 };
 ```
 
+Isolation and separation of responsibilities is generally hard to solve. The most recurrent scenario in a composable app with Ember Engines is: `Imagine the css or components that are only used by engine-a, engine-b and engine-c`  all possible cases have down sides here:
+
+ 1) Include shared deps into the host app - Downside: Increases initial bundle size.
+
+ 2) Include shared deps into each engine bundle - Downside: Duplicated stuff in multiple engine bundles.
+
+ 3) Somehow split out shared deps into dedicated bundles `(e.g. /engine-assets/shared-by-a-and-b.js)` - Downside: You might end up with 20 bundles `(e.g. shared-by-a-and-b, shared-by-b-and-c-, shared-by-a-b-and-c, ...)`, and it is probably pretty complicated from a built step perspective.
+
+The **case 2** is currently the best approach.
```

Comment:
Saying "best" here is a bit subjective. You could simply say "Engines currently take the second approach, but may allow more options in the future."

Reply:
fixed!